### PR TITLE
Re-use request pointer and context

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR addresses three HTTP request handling issues in custom elasticsearch-proxy handlers:
1. It replaces request copies by re-using the request pointer provided by `ServeHTTP`.
2. It replaces the custom request context implementation and copying by using the `http.Request` context.
3. Remove http2 transport configuration for upstream calls to Elasticsearch, because Elasticsearch does not support HTTP2 yet.
 
/cc @blockloop 
/assign @ewolinetz

### CODE REVIEW

**Comment by ewolinetz** [comment]
/hold
@periklis do you have any graphs showing the difference in the memory footprint during a load test before and after this change?

**Comment by periklis** [comment]
@ewolinetz @blockloop I am removing the BZ label from this PR as it is clearly not fixing the issue. However, it is a good enhancement for better maintainability.

**Comment by periklis** [comment]
@blockloop @ewolinetz Can we add this minor enhancement please?

File: pkg/handlers/authorization/handler.go [hunk 1]
```diff
@@ -50,58 +52,83 @@ func (auth *authorizationHandler) Name() string {
 
 //Process the request for authorization. The handler first attempts to get userinfo using bearer token
 //and falls back to the certificate subject or fails
-func (auth *authorizationHandler) Process(req *http.Request, context *handlers.RequestContext) (*http.Request, error) {
+func (auth *authorizationHandler) Process(req *http.Request) (*http.Request, error) {
```
Comment (blockloop) [inline pkg/handlers/authorization/handler.go:55]: I still don't understand the reason for returning the HTTP request at all... Can't you just remove this and only return error?

**Review by blockloop — APPROVED** [review]
/lgtm

File: pkg/proxy/server.go [hunk 2]
```diff
@@ -150,33 +141,33 @@ func (p *ProxyServer) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
...
-			p.StructuredError(responseWriter, err)
-			return
+			p.StructuredError(rw, err)
+			break
 		}
 	}
-	log.Debugf("Request: %v", alteredReq)
-	p.serveMux.ServeHTTP(responseWriter, alteredReq)
+
+	// log.Debugf("Request: %v", alteredReq)
```
Comment (jcantrill) [inline pkg/proxy/server.go:157]: Should either delete or uncomment

**Review by jcantrill — COMMENTED** [review]
/approve

one nit

**Review by jcantrill — COMMENTED** [review]
/lgtm

**Pull request merged**